lighthouse-core/fraggle-rock/api.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const _ = require('lodash');
+const Driver = require('./gather/driver.js');
+const Runner = require('../runner.js');
+const Config = require('../config/config.js');
+
+/**
+ * @param {LH.Gatherer.GathererInstance} gatherer
+ * @return {gatherer is LH.Gatherer.FRGathererInstance}
+ */
+function isFRGatherer(gatherer) {
+  // TODO(FR-COMPAT): use configuration on gatherer.meta to detect interface compatibility
+  return gatherer.name === 'Accessibility';
+}
+
+/** @param {{page: import('puppeteer').Page, config?: LH.Config.Json}} options */
+async function snapshot(options) {
+  const config = new Config(options.config);
+  const driver = new Driver(options.page);
+  await driver.connect();
+
+  const url = await options.page.url();
+
+  return Runner.run(
+    async () => {
+      /** @type {LH.BaseArtifacts} */
+      const baseArtifacts = {
+        fetchTime: new Date().toJSON(),
+        LighthouseRunWarnings: [],
+        URL: {requestedUrl: url, finalUrl: url},
+        Timing: [],
+        Stacks: [],
+        settings: config.settings,
+        // TODO(FR-COMPAT): convert these to regular artifacts
+        HostFormFactor: 'mobile',
+        TestedAsMobileDevice: true,
+        HostUserAgent: 'unknown',
```

Comment:
Since a breaking change is coming up, should we make some base artifacts optional? or nullable.

Reply:
I'm gathering the list and discussing at next FR product sync if we care about these concepts, if the answer is no, then I agree we'll try to make them optional before v7 for smoother transition 👍

Reply:
results of sync today were that we'll make these optional, the pending #10910 will take care of the device oriented ones naturally. since that's larger though, I'll save this for a followup PR 👍